visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -35,15 +33,15 @@ export class CodeMapArrowService
 		this.scale()
 	}
 
-	onBuildingSelected(selectedBuilding: CodeMapBuilding) {
-		if (this.isEdgeApplicableForBuilding(selectedBuilding)) {
+	onBuildingSelected = (data: { building: CodeMapBuilding }) => {
+		if (this.isEdgeApplicableForBuilding(data.building)) {
 			this.clearArrows()
 			this.showEdgesOfBuildings()
 		}
 		this.scale()
 	}
 
-	onBuildingDeselected() {
+	onBuildingDeselected = () => {
```

Comment:
Why did you choose an arrow function here? What is the advantage of it?

Reply:
the arrow function binds the `this` of the class to the function. In JS `this` is not bound to a class by default but derived from its calling context (a design flaw in my personal opinion). In case you are really curious https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this ;)